lib/engine/step/track.rb

```diff
@@ -39,18 +39,7 @@ def process_lay_tile(action)
       end
 
       def available_hex(entity, hex)
-        connected = hex_neighbors(entity, hex)
-        return nil unless connected
-
-        tile_lay = get_tile_lay(entity)
-        return nil unless tile_lay
-
-        color = hex.tile.color
-        return nil if color == :white && !tile_lay[:lay]
-        return nil if color != :white && !tile_lay[:upgrade]
-        return nil if color != :white && tile_lay[:cannot_reuse_same_hex] && @round.laid_hexes.include?(hex)
-
-        connected
+        base_available_hex(entity, hex)
```

Comment:
Why this change?

Reply:
tracker.rb which step uses doesn't have this code and 18ireland needs to effectively use it for the Board of Works private.

i'm not entirely sure about the naming, hopefully Toby can have a look

Reply:
@tobymao could do with some opinions

Reply:
Does `reachable: true` on the ability definition solve this? From the description in the abilities README it sounds like it raises an error after placing the tile, so I don't know if it affects which hexes are highlighted (which I think is what you're trying to accomplish here?)

Maybe the `Engine::Step::SpecialTrack` available_hex logic could be modified so that if `reachable` is true, then only reachable hexes are highlighted? (i.e., generalize what you have in `G18Ireland::Step::SpecialTrack#available_hex`)

Instead of `base_available_hex`, maybe `tracker_available_hex`?

Reply:
i need more context? can you hit me up on slack about it?

Reply:
@michaeljb unfortunatly you can't do that generalization as this is effectively a player choice for a discount on terrain costs, but is done using their normal placement rules. 1822 uses reachable without hexes to mean an extra lay.

We could add an attribute which would be 'counts towards lays' which would also tidy up 1817's privates a bit

Reply:
I'm probably gonna rework this after some mild refactoring in 1817 to make this a lot easier.

Reply:
should be good now.